### ISSUE TITLE
Problemes al Train_classifier

### ISSUE THREAD

**Issue by gerardkm94** [gdsa-upc/gdsa-suport#28]
Quan utilitzem les funcions de "SVM" pel "Classifier", més concretament el "clf.fit", ens sorgeix un error.
Al passarli les annotacions del train set i els labels, em retorna un error que diu:
X and Y (anotacions y labels) tenen formes incompatibles, X has 223395 samples and y has 451(perquè no hem tret les classes repetides per provar, quan les traiem aquest número serà 13).

Tens idea de per què pot passar?

Gràcies!

**Comment by amaiasalvador** [comment]
Hola Gerard,

al mètode _fit()_ de la classe [sklearn.svm.SVC](http://scikit-learn.org/stable/modules/generated/sklearn.svm.SVC.html#sklearn.svm.SVC.fit) li heu de passar:

1) _X_: Els descriptors BoW de les 450 imatges d'entrenament. Si el vostre vocabulari era de 100 centroides, _X_ serà de tamany (450,100).
2) _y_: Una llista amb les anotacions per les imatges d'entrenament. Cada posició _i_ de la llista _y_ haurà de contenir el nom (o la id) de la classe a la qual pertany cada imatge d'entrenament (representada en _X_ pel descriptor BoW en la posició X[_i_,:] ). La longitud de _y_ ha de ser igual al número de mostres que hi ha a _X_: 450. D'aquesta manera, no heu d'eliminar les repeticions, ja que el classificador necessita saber la classe a la qual pertany cada descriptor BoW que hi ha a X.

Mirant el tamany de X que em dius (~200.000 samples), crec que en comptes de fer servir els descriptors BoW esteu fent servir els descriptors locals de totes les imatges d'entrenament. No entenc però per què us surt la _y_ de tamany 451, hauria de ser 450... Reviseu això.
